staff/views/coupon.py

```diff
@@ -34,21 +34,37 @@ def get(self, request):
     def post(self, request):
         coupon_form = CouponForm(request.POST)
 
-        if coupon_form.is_valid():
-            try:
-                with transaction.atomic():
-                    coupon = coupon_form.save()
-
-                    return redirect('coupon_detail', coupon_id=coupon.id)
-            except IntegrityError:
-                return redirect('coupon_new')
-        return render(
-            request,
-            'coupon/new.html',
-            {
-                'coupon_form': coupon_form
-            }
-        )
+        if not coupon_form.is_valid():
+            return render(
+                request,
+                'coupon/new.html',
+                {
+                    'coupon_form': coupon_form
+                }
+            )
+
+        code = coupon_form.cleaned_data.get('code')
+        if Coupon.objects.filter(code=code).exists():
+            coupon_form.add_error(
+                'code',
+                f"Coupon with code {code} already exists"
+            )
+
```

Comment:
Maybe we could add comments and adjust spacing in this file to group related logic together and increase readability?